langchain/src/chat_models/iflytek_xinghuo/common.ts

```diff
@@ -0,0 +1,479 @@
+   * multiple choice, and temp closer to 1 for creative
+   * and generative tasks. Defaults to 0.5.
+   */
+  temperature?: number;
+
+  max_tokens?: number;
+
+  top_k?: number;
+
+  streaming?: boolean;
+}
+
+/**
+ * Wrapper around IflytekXingHuo large language models that use the Chat endpoint.
+ *
+ * To use you should have the `IFLYTEK_API_KEY` and `IFLYTEK_API_SECRET` and `IFLYTEK_APPID`
+ * environment variable set.
+ *
+ * @augments BaseChatModel
+ * @augments IflytekXinghuoChatInput
+ */
+export abstract class BaseChatIflytekXinghuo
+  extends BaseChatModel
+  implements IflytekXinghuoChatInput
+{
+  static lc_name() {
+    return "ChatIflytekXinghuo";
+  }
+
+  get callKeys(): string[] {
+    return ["stop", "signal", "options"];
+  }
+
+  get lc_secrets(): { [key: string]: string } | undefined {
+    return {
+      iflytekApiKey: "IFLYTEK_API_KEY",
+      iflytekApiSecret: "IFLYTEK_API_SECRET",
+    };
+  }
+
+  get lc_aliases(): { [key: string]: string } | undefined {
+    return undefined;
+  }
+
+  lc_serializable = true;
+
+  version = "v2.1";
+
+  iflytekAppid: string;
+
+  iflytekApiKey: string;
+
+  iflytekApiSecret: string;
+
+  userId?: string;
+
+  apiUrl: string;
+
+  domain: string;
+
+  temperature = 0.5;
+
+  max_tokens = 2048;
+
+  top_k = 4;
+
+  streaming = false;
+
+  constructor(fields?: Partial<IflytekXinghuoChatInput> & BaseChatModelParams) {
+    super(fields ?? {});
+
+    const iflytekAppid =
+      fields?.iflytekAppid ?? getEnvironmentVariable("IFLYTEK_APPID");
+    if (!iflytekAppid) {
+      throw new Error("Iflytek APPID not found");
+    } else {
+      this.iflytekAppid = iflytekAppid;
+    }
+
+    const iflytekApiKey =
+      fields?.iflytekApiKey ?? getEnvironmentVariable("IFLYTEK_API_KEY");
+    if (!iflytekApiKey) {
+      throw new Error("Iflytek API key not found");
+    } else {
+      this.iflytekApiKey = iflytekApiKey;
+    }
+
+    const iflytekApiSecret =
+      fields?.iflytekApiSecret ?? getEnvironmentVariable("IFLYTEK_API_SECRET");
+    if (!iflytekApiSecret) {
+      throw new Error("Iflytek API secret not found");
+    } else {
+      this.iflytekApiSecret = iflytekApiSecret;
+    }
+
+    this.userId = fields?.userId ?? this.userId;
+    this.streaming = fields?.streaming ?? this.streaming;
+    this.temperature = fields?.temperature ?? this.temperature;
+    this.max_tokens = fields?.max_tokens ?? this.max_tokens;
+    this.top_k = fields?.top_k ?? this.top_k;
+
+    this.version = fields?.version ?? this.version;
+    if (["v1.1", "v2.1", "v3.1"].includes(this.version)) {
+      switch (this.version) {
+        case "v1.1":
+          this.domain = "general";
+          break;
+        case "v2.1":
+          this.domain = "generalv2";
+          break;
+        case "v3.1":
+          this.domain = "generalv3";
+          break;
+        default:
+          this.domain = "generalv2";
+      }
+      this.apiUrl = `wss://spark-api.xf-yun.com/${this.version}/chat`;
```

Comment:
Do you have a source for this API URL?

Reply:
yes, I have an official document site. But it's writed by chinese. The english document is not updated.
the site: https://www.xfyun.cn/doc/spark/Web.html

![image](https://github.com/langchain-ai/langchainjs/assets/9697715/7d42a71a-b7c7-4f45-8040-57d664d2d54e)